cluster/local/integration_tests.sh

```diff
@@ -0,0 +1,234 @@
+K8S_CLUSTER="${K8S_CLUSTER:-${BUILD_REGISTRY}-INTTESTS}"
+
+CROSSPLANE_NAMESPACE="crossplane-system"
+PACKAGE_NAME="provider-gcp"
+
+# cleanup on exit
+if [ "$skipcleanup" != true ]; then
+  function cleanup {
+    echo_step "Cleaning up..."
+    export KUBECONFIG=
+    "${KIND}" delete cluster --name="${K8S_CLUSTER}"
+  }
+
+  trap cleanup EXIT
+fi
+
+echo_step "creating k8s cluster using kind"
+"${KIND}" create cluster --name="${K8S_CLUSTER}"
+
+# tag package and controller images and load then into kind cluster
+docker tag "${BUILD_IMAGE}" "${PACKAGE_IMAGE}"
+docker tag "${CONTROLLER_IMAGE}" "${PACKAGE_CONTROLLER_IMAGE}"
+"${KIND}" load docker-image "${PACKAGE_IMAGE}" --name="${K8S_CLUSTER}"
+"${KIND}" load docker-image "${PACKAGE_CONTROLLER_IMAGE}" --name="${K8S_CLUSTER}"
+
+# wait for kind pods
+echo_step "wait for kind pods"
+kindpods=("kube-apiserver-${BUILD_REGISTRY}-inttests-control-plane" "kube-controller-manager-${BUILD_REGISTRY}-inttests-control-plane" "kube-scheduler-${BUILD_REGISTRY}-inttests-control-plane")
+wait_for_pods_in_namespace 120 "kube-system" "${kindpods[@]}"
+
+# install crossplane from master channel
+echo_step "installing crossplane from master channel"
+"${KUBECTL}" create ns crossplane-system
+"${HELM3}" repo add crossplane-master https://charts.crossplane.io/master/
+chart_version="$("${HELM3}" search repo crossplane-master/crossplane --devel | awk 'FNR == 2 {print $2}')"
+echo_info "using crossplane version ${chart_version}"
+echo
+"${HELM3}" install crossplane --namespace crossplane-system crossplane-master/crossplane --version ${chart_version} --devel
+
+echo_step "waiting for deployment crossplane rollout to finish"
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" rollout status "deploy/crossplane" --timeout=2m
+
+echo_step "wait until the pods are up and running"
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" wait --for=condition=Ready pods --all --timeout=1m
+
+# ----------- integration tests
+echo_step "--- INTEGRATION TESTS ---"
+echo
+echo_step "check for necessary deployment statuses"
+echo
+echo -------- deployments
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" get deployments
+
+check_deployments "crossplane" "${CROSSPLANE_NAMESPACE}"
+
+echo_step "check for crossplane pods statuses"
+echo
+echo "--- pods ---"
+check_pods $("${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" get pods)
+
+# install package
+echo_step "installing ${PROJECT_NAME} into \"${CROSSPLANE_NAMESPACE}\" namespace"
+
+INSTALL_YAML="$( cat <<EOF
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: Provider
+metadata:
+  name: "${PACKAGE_NAME}"
+spec:
+  package: "${PACKAGE_IMAGE}"
+EOF
+)"
+
+echo "${INSTALL_YAML}" | "${KUBECTL}" apply -f -
+
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" get deployments
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" get pods
+
+# this is to let package manager unpack pods and start the controller. If for
+# some reason the package manager did not create these pods at all, then this
+# could give a false positive.
```

Comment:
Just calling this out. These integration tests are already not very robust, so I don't see this as much of a regression, but I would like to introduce more robust integration tests (and ones that are not bash files) _very_ soon.